DTS-STN/canadian-dental-care-plan
# Adding modules to make letter type service injectable

### Description
Also removed `getAllLetterTypes(..)` from `letters-service`, which cannot be injected, and changed these references to the new `letter-type.service`.

### Screenshots (if applicable)
![image](https://github.com/user-attachments/assets/5cb55964-995c-411d-b18f-311f7bb96bd9)

### Checklist

### Test Instructions
Run the application and navigate to `/en/letters`. Verify that the page works as before and you can view letters with letter types displayed.

### Additional Notes
`letter-types.json` will be moved to the correct `.server` directory in a future PR - would have added an extra 500+ lines to this PR.
